### PR TITLE
Change net logger

### PR DESCRIPTION
**Pull request by vadimsmirnovnsk**
It's just a **draft** PR because I haven't finished building the environment for Embox developing yet.
Code has been written just in Sublime.
Build has been completed with `make confload-arm/qemu`.

Please check at first, is it what did you mean)
And please don't be afraid to make cruel CR's)

### CODE REVIEW

**Comment by abusalimov** [comment]
Thanks for your contribution!

Regarding your commits:
 - Please linearize your branch, there should be no unnecessary merge commits. You will need to rebase, here is how you do it: https://github.com/embox/embox/wiki/Contributing:-Git-cheatsheet
 - (minor) If possible, please, be more specific and explicit in your commit messages: https://github.com/embox/embox/blob/721-contributing-guide/CONTRIBUTING.md#commit-message-agreements

File: src/net/l0/net_rx.c [hunk 1]
```diff
@@ -33,7 +33,7 @@ int net_rx(struct sk_buff *skb) {
 	assert(skb != NULL);
 	assert(skb->dev != NULL);
 	if (skb->len < skb->dev->hdr_len) {
-		DBG(printk("net_rx: %p invalid length %zu\n", skb,
+		DBG(log_error("net_rx: %p invalid length %zu\n", skb,
```
Comment (abusalimov) [inline src/net/l0/net_rx.c]: Regarding the code: shouldn't all these occurrences (`DBG(log_xxx(..))`) be changed to a plain `log_debug` call? That is, to get rid of these nasty conditional macros eventually. @anton-bondarev what do you think?

Comment (anton-bondarev) [inline src/net/l0/net_rx.c]: Of cause, I'm agree. DBG macro should switch to log_error. And in Mybuild file you should add option "log_level"

Comment (vadimsmirnovnsk) [inline src/net/l0/net_rx.c]: Ok, will make it properly soon.

**Comment by vadimsmirnovnsk** [comment]
Sorry, I never worked with upstream before.
Now brunch has been rebased, and I fixed issues with adding `log_level` to `mybuild` files.
I haven't completed building environment so far, and branch tested only just by building project with `make confload-arm/qemu`.

**Comment by abusalimov** [comment]
Just push it. Travis will test few more templates, inc. x86

Regards,
Eldar

**Comment by abusalimov** [comment]
Ah, just noticed your commit.

File: src/net/l3/rarp.c [hunk 2]
```diff
@@ -147,32 +141,30 @@ static int rarp_hnd_request(const struct arphdr *rarph,
...
-			assert(rarph->ar_hln == ETH_ALEN);
-			printk("[" MACADDR_FMT "]",
-				MACADDR_FMT_ARG(dst_haddr));
-		}
-		else {
-			printk("[unknown(%x)]", htons(rarph->ar_hrd));
-		}
-		printk("\n");
-	});
+	log_info("rarp_hnd_request: send reply with ");
```
Comment (abusalimov) [inline src/net/l3/rarp.c]: Is such change appropriate?

Anyway, `log_info` is obviously too severe for debug output.

Comment (anton-bondarev) [inline src/net/l3/rarp.c]: I'm agree. log_debug() more appropriate for this line

**Comment by anton-bondarev** [comment]
LGTM

File: src/net/l3/rarp.c (hunk 2 above)
Comment (vadimsmirnovnsk) [inline src/net/l3/rarp.c]: Replaced by log_debug.

**Comment by anton-bondarev** [comment]
Yes, you shold if it's not difficult. The code will became better if you do so

**Comment by vadimsmirnovnsk** [comment]
Ok, completed with uncommenting some code lines)

File: src/net/l4/tcp.c [hunk 3]
```diff
@@ -54,12 +56,6 @@ EMBOX_NET_PROTO(ETH_P_IPV6, IPPROTO_TCP, tcp_rcv,
 #define MODOPS_VERIFY_CHKSUM OPTION_GET(BOOLEAN, verify_chksum)
 
 #define TCP_DEBUG 0
```
Comment (anton-bondarev) [inline src/net/l4/tcp.c]: This macro is used yet?

File: src/net/Mybuild [hunk 4]
```diff
@@ -1,25 +1,27 @@
 package embox.net
 
 module dev {
-	option number netdev_quantity=8
```
Comment (anton-bondarev) [inline src/net/Mybuild]: Please, don't touch a code if it's not necessary.
You swap some lines in this commit, can you rollback them?

File: src/net/l3/arp.c [hunk 5]
```diff
@@ -156,32 +150,30 @@ static int arp_hnd_request(const struct arphdr *arph,
...
-			assert(arph->ar_hln == ETH_ALEN);
-			printk("[" MACADDR_FMT "]",
-				MACADDR_FMT_ARG(dst_haddr));
-		}
-		else {
-			printk("[unknown(%x)]", htons(arph->ar_hrd));
-		}
-		printk("\n");
-	});
+	log_debug("arp_hnd_request: send reply with ");
```
Comment (anton-bondarev) [inline src/net/l3/arp.c]: Can you extract blocks like that in a routine?
You should use OPTION_GET(NUMBER,log_level) macro

File: src/net/l4/tcp.c (hunk 3 above)
Comment (vadimsmirnovnsk) [inline src/net/l4/tcp.c]: Yes,
`packet_print` depends of `debug_print`, and both depend of `TCP_DEBUG` macro.
I just changed `printf` function by `log_debug` in `debug_print` function, but I can reduce `TCP_DEBUG` by changing all `packet_print` and `debug_print` directly by `log_debug`.
Schould I do it?

File: src/net/Mybuild (hunk 4 above)
Comment (vadimsmirnovnsk) [inline src/net/Mybuild]: Sure, I can rollback it.
I just sorted it in alphabetical order.
Do you have any code style for ordering code lines in Mybuild files?

File: src/net/l4/tcp.c (hunk 3 above)
Comment (anton-bondarev) [inline src/net/l4/tcp.c]: Yes, Do it/ it will great

File: src/net/Mybuild (hunk 4 above)
Comment (anton-bondarev) [inline src/net/Mybuild]: Hm, Unfortunately haven't now. And we try to do changes only for branch task.

**Pull request merged**